### PR TITLE
Cypress/E2E: Fix flaky tests

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Fix flaky tests
- Clean up on unnecessary update in server config for `EnableBotAccountCreation` when true since it's the same as the default value

#### Ticket Link
none

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by mattermost-quality** [comment]
E2E test ongoing. See link to [dashboard](https://automation-dashboard.vercel.app/cycle/288090) and/or [pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/288090).

**Comment by mattermost-quality** [comment]
E2E result: 9 of 1605 tests failed :x:. See [link to report](https://automation-dashboard.vercel.app/cycle/288090).

**Comment by mattermost-quality** [comment]
E2E test ongoing. See link to [dashboard](https://automation-dashboard.vercel.app/cycle/288231) and/or [pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/288231).

**Comment by mattermost-quality** [comment]
E2E result: 12 of 1605 tests failed :x:. See [link to report](https://automation-dashboard.vercel.app/cycle/288231).

File: e2e/cypress/tests/integration/plugins/plugin_buttons_spec.js [hunk 1]
```diff
@@ -20,26 +20,13 @@ describe('collapse on 15 plugin buttons', () => {
...
         // # Uninstall all plugins
-        cy.apiGetAllPlugins().then(({plugins}) => {
-            const {active, inactive} = plugins;
-            inactive.forEach((plugin) => cy.apiRemovePluginById(plugin.id));
-            active.forEach((plugin) => cy.apiRemovePluginById(plugin.id));
-        });
+        cy.apiUninstallAllPlugins();
     });
 
     it('MM-T1649 Greater than 15 plugin buttons collapse to one icon in top nav', () => {
```
Comment (saturninoabril) [inline e2e/cypress/tests/integration/plugins/plugin_buttons_spec.js:32]: Here we'll still need to find a way on how to test with Boards product enabled. Currently it failed on Boards product but not with Boards plugin. Will deal with it in separate PR.

**Review by nevyangelova — APPROVED** [review]
LGTM :)

**Review by furqanmlk — APPROVED** [review]
Thanks @saturninoabril,  LGTM

**Pull request merged**